core/src/Streamly/Internal/Data/Stream/Serial/Eliminate.hs

```diff
@@ -0,0 +1,21 @@
+module Streamly.Internal.Data.Stream.Serial.Eliminate
+    (
+      drain
+    )
+where
+
+#if __GLASGOW_HASKELL__ < 808
+import Data.Semigroup (Semigroup(..))
+#endif
+
+import qualified Streamly.Internal.Data.Stream.Common as P
+
+import Streamly.Internal.Data.Stream.Serial.Type
+
+import Prelude hiding (map, mapM, repeat, filter)
+
+#include "inline.hs"
+
+{-# INLINE drain #-}
+drain :: Monad m => SerialT m a -> m ()
```

Comment:
Add the same documentation as in Streamly.Prelude (i.e. Streamly.Internal.Data.Stream.IsStream).
Add all the docs in a separate commit.